storage.go

```diff
@@ -620,18 +620,20 @@ func (s *Storage) stat(ctx context.Context, path string, opt pairStorageStat) (o
 	o.ID = rp
 	o.Path = path
 
-	metadata := output.Metadata
-	if target, ok := metadata["x-amz-meta-bs-link-target"]; ok {
-		// The path is a symlink object.
-		if !s.features.VirtualLink {
-			err = NewOperationNotImplementedError("virtual_link")
-			return nil, err
-		}
+	if output.Metadata != nil {
+		metadata := output.Metadata
+		if target, ok := metadata["X-Amz-Meta-Bs-Link-Target"]; ok {
+			// The path is a symlink object.
+			if !s.features.VirtualLink {
```

Comment:
If virtual link is not enabled, we should treat it as a normal object instead return error directly.

Reply:
Got it!

Reply:
> If virtual link is not enabled, we should treat it as a normal object instead return error directly.

I'm a bit confused, what is the mode of this normal object set to?
Or even if you don't have virtual link enabled, you can still set it to link mode?

Reply:
Virtual link not enabled means no simulated symlink for the service, so all the object mode should not be `ModeLink`.  Treat as a normal object means we should continue the assignment for the returning object here and check whether the object is a nomal object(ModeRead) or a simulated dir(ModeDir).

Reply:
> Virtual link not enabled means no simulated symlink for the service, so all the object mode should not be `ModeLink`. Treat as a normal object means we should continue the assignment for the returning object here and check whether the object is a nomal object(ModeRead) or a simulated dir(ModeDir).

Got it!